pyomo/core/kernel/component_interface.py

```diff
@@ -210,6 +212,8 @@ def __str__(self):
             for a way to generate a static set of component
             names.
         """
+        if ICategorizedObject.labeler:
```

Comment:
That makes total sense, and as a relative newbie to pyomo I'm all for more documentation.